client/blocks/author-selector/switcher-shell.jsx

```diff
@@ -180,6 +181,10 @@ class AuthorSwitcherShell extends React.Component {
 
 	renderAuthor = author => {
 		const authorGUID = this.getAuthorItemGUID( author );
+		if ( this.props.transformAuthor ) {
+			author = this.props.transformAuthor( author );
```

Comment:
it would be really great if we didn't muck around with the input arguments to this (or any) function. it's pretty confusing to me when I come around and see a function's declaration and think I have one thing, like the `author`, but then discover later that due to a conditional I have another thing.

would you consider creating a new binding and preserving the argument?

```js
renderAuthor = rawAuthor => {
	const { transformAuthor } = this.props;

	const authorGUID = this.getAuthorItemGUID( rawAuthor );
	const author = transformAuthor ? transformAuthor( rawAuthor ) : rawAuthor;

	…
}
```